pkg/cmd/util/clientcmd/factory_client_access.go

```diff
@@ -51,7 +63,22 @@ type ring0Factory struct {
 type ClientAccessFactory interface {
 	kcmdutil.ClientAccessFactory
 
+	// TODO: this should be removed when we finally get rid of pkg/client
 	Clients() (*client.Client, kclientset.Interface, error)
+	// use below clients instead whenever needed:
+	OpenShiftAppsClient() (appsclient.Interface, error)
```

Comment:
nit: I would rather have this as `OpenshiftInternalAppsClient` to be consistent with https://github.com/mfojtik/origin/blob/master/pkg/cmd/server/origin/controller/interfaces.go#L60